lib/Dialect/AffineOps/AffineOps.cpp

```diff
@@ -1354,11 +1354,53 @@ struct AffineForLoopBoundFolder : public OpRewritePattern<AffineForOp> {
     return matchSuccess();
   }
 };
+
+// This is a pattern to canonicalize affine for op loop bounds.
+struct AffineForOpBoundCanonicalizer : public OpRewritePattern<AffineForOp> {
+  using OpRewritePattern<AffineForOp>::OpRewritePattern;
+
+  PatternMatchResult matchAndRewrite(AffineForOp forOp,
+                                     PatternRewriter &rewriter) const override {
+    SmallVector<Value *, 4> lbOperands(forOp.getLowerBoundOperands());
```

Comment:
How about this?

```
SmallVector<Value *, 4> lbOperands(forOp.getLowerBoundOperands());
SmallVector<Value *, 4> ubOperands(forOp.getUpperBoundOperands());

auto lbMap = forOp.getLowerBoundMap();
auto ubMap = forOp.getUpperBoundMap();
auto tmpLbMap = lbMap;
auto tmpUbMap = ubMap;

canonicalizeMapAndOperands(&lbMap, &lbOperands);
canonicalizeMapAndOperands(&ubMap, &ubOperands);

// If none of lbMap and ubMap are update by canonicalization, 
// don't rewrite.
if (lbMap == tmpLbMap && ubMap == tmpUbMap)
  return matchFailure();
      
forOp.setLowerBound(lbOperands, lbMap);
forOp.setUpperBound(ubOperands, ubMap);
rewriter.updatedRootInPlace(forOp);
return matchSuccess();
```

Reply:
Yes, this works too, and is simpler - but we don't need to update both lb and ub operands if only one of them had been changed. Updated to it with a slight change. Thanks.